kapacitor/client.go

```diff
@@ -259,6 +268,11 @@ func (c *Client) Update(ctx context.Context, href string, rule chronograf.AlertR
 		return nil, err
 	}
 
+	prevStatus, err := c.status(ctx, href)
```

Comment:
I think this races with the write on L296. If you have two fast update requests where the first is blocked on L289 for longer than the second one, I think it's possible for the task to end up in a disabled state. We should probably drop a `sync.Mutex` here so that updates happen atomically.